src/goto-programs/label_function_pointer_call_sites.cpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************\
+Module: Label function pointer call sites
+Author: Diffblue Ltd.
+\*******************************************************************/
+
+/// \file
+/// Label function pointer call sites across a goto model
+
+#include "label_function_pointer_call_sites.h"
+
+#include <util/fresh_symbol.h>
+
+void label_function_pointer_call_sites(goto_modelt &goto_model)
+{
+  for(auto &goto_function : goto_model.goto_functions.function_map)
+  {
+    std::size_t function_pointer_call_counter = 0;
+    for_each_instruction_if(
+      goto_function.second,
+      [](const goto_programt::targett it) {
+        return it->is_function_call() && can_cast_expr<dereference_exprt>(
+                                           it->get_function_call().function());
+      },
+      [&](goto_programt::targett it)
+      {
+        auto const &function_call = it->get_function_call();
+        auto const &function_pointer_dereference =
+          to_dereference_expr(function_call.function());
+        auto const &source_location = function_call.source_location();
+        auto const &goto_function_symbol_mode =
+          goto_model.symbol_table.lookup_ref(goto_function.first).mode;
+
+        auto const call_site_symbol_name =
+          irep_idt{id2string(goto_function.first) + ".function_pointer_call." +
```

Comment:
May be best not to use `.` since it delimits members, and this isn't one ($ has been used elsewhere for a reserved character that doesn't mean anything in C)

Reply:
Agreed. This will change in the next PR anyways so I'll leave this as is for now.